aspnetcore/fundamentals/environments.md

```diff
@@ -242,13 +242,121 @@ See [Configuration by environment](xref:fundamentals/configuration/index#configu
 
 ## Environment-based Startup class and methods
 
-When an ASP.NET Core app starts, the [Startup class](xref:fundamentals/startup) bootstraps the app. If a `Startup{EnvironmentName}` class exists, the class is called for that `EnvironmentName`:
+### Startup class conventions
 
-[!code-csharp[](environments/sample/EnvironmentsSample/StartupDev.cs?name=snippet&highlight=1)]
+When an ASP.NET Core app starts, the [Startup class](xref:fundamentals/startup) bootstraps the app. The app can define separate `Startup` classes for different environments (for example, `StartupDevelopment`), and the appropriate `Startup` class is selected at runtime. The class whose name suffix matches the current environment is prioritized. If a matching `Startup<EnvironmentName>` class isn't found, the `Startup` class is used.
 
-[WebHostBuilder.UseStartup&lt;TStartup&gt;](/dotnet/api/microsoft.aspnetcore.hosting.webhostbuilderextensions.usestartup#Microsoft_AspNetCore_Hosting_WebHostBuilderExtensions_UseStartup__1_Microsoft_AspNetCore_Hosting_IWebHostBuilder_) overrides configuration sections.
+To implement environment-based `Startup` classes, create a `Startup<EnvironmentName>` class for each environment in use and a fallback `Startup` class:
```

Comment:
The use of angle brackets here is confusing, it looks like generics. Is there another seperator you can use? `|` (pipe)? `Startup|EnvironmentName|`

Reply:
I see what you mean. We use braces from time-to-time, and Rick did braces for this topic back in the day.

`Startup{EnvironmentName}`

Reply:
Good enough.